### PR TITLE
Upgrade to Node-18 and NG-15

### PR DESCRIPTION
**Pull request by codaimaster**
**JIRA link (if applicable)**
https://tools.hmcts.net/jira/browse/EXUI-763

**Change description**
Upgrade to Node 18 & NG-15

**Does this PR introduce a breaking change? (check one with "x")**
[ ] Yes
[X] No

### CODE REVIEW

File: projects/ccd-case-ui-toolkit/src/lib/shared/components/case-viewer/case-review-specific-access-request/case-review-specific-access-request.component.spec.ts [hunk 1]
```diff
@@ -49,12 +50,13 @@ class MockActivatedRouteSnapshot implements ActivatedRouteSnapshot {
 class MockActivatedRoute implements ActivatedRoute {
   public snapshot: ActivatedRouteSnapshot;
   public url: Observable<UrlSegment[]>;
+  public title: Observable<string>;
   public params: Observable<Params>;
   public queryParams: Observable<Params>;
   public fragment: Observable<string>;
   public data: Observable<Data>;
   public outlet: string;
-  public component: Type<any> | string;
```
Comment (codaimaster) [inline projects/ccd-case-ui-toolkit/src/lib/shared/components/case-viewer/case-review-specific-access-request/case-review-specific-access-request.component.spec.ts:57]: Check implication, pls.

File: projects/ccd-case-ui-toolkit/src/lib/shared/components/case-viewer/case-viewer.component.spec.ts [hunk 2]
```diff
@@ -104,12 +105,13 @@ class MockActivatedRouteSnapshot implements ActivatedRouteSnapshot {
 class MockActivatedRoute implements ActivatedRoute {
   public snapshot: ActivatedRouteSnapshot;
   public url: Observable<UrlSegment[]>;
+  public title: Observable<string>;
   public params: Observable<Params>;
   public queryParams: Observable<Params>;
   public fragment: Observable<string>;
   public data: Observable<Data>;
   public outlet: string;
-  public component: Type<any> | string;
```
Comment (codaimaster) [inline projects/ccd-case-ui-toolkit/src/lib/shared/components/case-viewer/case-viewer.component.spec.ts:112]: Check implication, pls.

File: projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/case-flag/read-case-flag-field.component.ts [hunk 3]
```diff
@@ -53,9 +53,9 @@ export class ReadCaseFlagFieldComponent extends AbstractFieldReadComponent imple
       this.caseLevelCaseFlagData = this.flagsData.find(
         instance => instance.pathToFlagsFormGroup === this.caseLevelCaseFlagsFieldId);
     } else if (this.context === PaletteContext.CHECK_YOUR_ANSWER) {
-      // If the context is PaletteContext.CHECK_YOUR_ANSWER, the Flags data is already present within the FormGroup.
+      // If the context is PaletteContext.CHECK_YOUR_ANSWER, the Flags data is already present within the formGroup.
```
Comment (codaimaster) [inline projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/case-flag/read-case-flag-field.component.ts:56]: Fix the case pls.

File: projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/complex/read-complex-field.component.spec.ts [hunk 4]
```diff
@@ -31,7 +31,7 @@ describe('ReadComplexFieldComponent', () => {
 
     readComplexFieldNewTableComponentMock = MockComponent({
       selector: 'ccd-read-complex-field-collection-table',
-      inputs: ['caseField', 'context', 'formGroup', 'topLevelFormGroup', 'idPrefix']
+      inputs: ['caseField', 'context', 'UntypedFormGroup', 'topLevelFormGroup', 'idPrefix']
```
Comment (codaimaster) [inline projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/complex/read-complex-field.component.spec.ts:34]: Fix please.

Comment (codaimaster) [reply projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/complex/read-complex-field.component.spec.ts:34]: UntypedFormGroup -> formGroup

File: projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/complex/write-complex-field.component.spec.ts [hunk 5]
```diff
@@ -28,15 +28,15 @@ describe('WriteComplexFieldComponent', () => {
 
   const fieldWriteComponentMock = MockComponent({
     selector: 'ccd-field-write',
-    inputs: ['caseField', 'caseFields', 'formGroup', 'idPrefix', 'isExpanded', 'parent', 'isInSearchBlock']
+    inputs: ['caseField', 'caseFields', 'UntypedFormGroup', 'idPrefix', 'isExpanded', 'parent', 'isInSearchBlock']
```
Comment (codaimaster) [inline projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/complex/write-complex-field.component.spec.ts:31]: UntypedFormGroup -> formGroup

File: projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/complex/write-complex-field.component.spec.ts [hunk 6]
```diff
@@ -231,7 +231,7 @@ describe('WriteComplexFieldComponent', () => {
       expect(labels[LINE_2].componentInstance.caseField.label).toBe(FIELD_TYPE_WITH_VALUES.complex_fields[LINE_2].label);
     });
 
-    it('should return control if it exists in the formGroup', () => {
+    it('should return control if it exists in the UntypedFormGroup', () => {
```
Comment (codaimaster) [inline projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/complex/write-complex-field.component.spec.ts:234]: UntypedFormGroup -> formGroup

File: projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/complex/write-complex-field.component.spec.ts [hunk 7]
```diff
@@ -251,7 +251,7 @@ describe('WriteComplexFieldComponent', () => {
     });
 
     // TODO: Ensure there is an equivalent test for AbstractFormFieldComponent.register.
-    it('should add control if it does not exist in the formGroup', () => {
+    it('should add control if it does not exist in the UntypedFormGroup', () => {
```
Comment (codaimaster) [inline projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/complex/write-complex-field.component.spec.ts:254]: UntypedFormGroup -> formGroup

File: projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/date/write-date-field.component.spec.ts [hunk 8]
```diff
@@ -62,7 +64,7 @@ describe('WriteDateFieldComponent', () => {
     fixture.detectChanges();
   }));
 
-  it('should add a formControl linked to the field ID to the formGroup', () => {
+  it('should add a formControl linked to the field ID to the UntypedFormGroup', () => {
```
Comment (codaimaster) [inline projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/date/write-date-field.component.spec.ts:67]: UntypedFormGroup -> formGroup

File: projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/email/write-email-field.component.spec.ts [hunk 9]
```diff
@@ -66,7 +68,7 @@ describe('WriteEmailFieldComponent', () => {
     fixture.detectChanges();
   }));
 
-  it('should add a formControl linked to the field ID to the formGroup', () => {
+  it('should add a formControl linked to the field ID to the UntypedFormGroup', () => {
```
Comment (codaimaster) [inline projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/email/write-email-field.component.spec.ts:71]: UntypedFormGroup -> formGroup

File: projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/judicial-user/write-judicial-user-field.component.ts [hunk 10]
```diff
@@ -29,18 +29,18 @@ export class WriteJudicialUserFieldComponent extends WriteComplexFieldComponent
...
     super(compoundPipe, validatorsService);
   }
 
   public ngOnInit(): void {
     super.ngOnInit();
     this.judicialUserControl = new FormControl(this.caseField.value);
-    // FormControl needs to be added to the main FormGroup so it can be picked up by the PageValidationService when
-    // checking if the page is valid. FormGroup.setControl() is used here to ensure any existing JudicialUser
+    // FormControl needs to be added to the main UntypedFormGroup so it can be picked up by the PageValidationService when
+    // checking if the page is valid. formGroup.setControl() is used here to ensure any existing JudicialUser
```
Comment (codaimaster) [inline projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/judicial-user/write-judicial-user-field.component.ts:43]: fix casing

File: projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/money-gbp/write-money-gbp-field.component.spec.ts [hunk 11]
```diff
@@ -60,7 +60,7 @@ describe('WriteMoneyGbpFieldComponent', () => {
     fixture.detectChanges();
   }));
 
-  it('should add a formControl linked to the field ID to the formGroup', () => {
+  it('should add a formControl linked to the field ID to the UntypedFormGroup', () => {
```
Comment (codaimaster) [inline projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/money-gbp/write-money-gbp-field.component.spec.ts:63]: UntypedFormGroup -> formGroup

File: projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/number/write-number-field.component.spec.ts [hunk 12]
```diff
@@ -66,7 +68,7 @@ describe('WriteNumberFieldComponent', () => {
     fixture.detectChanges();
   }));
 
-  it('should add a formControl linked to the field ID to the formGroup', () => {
+  it('should add a formControl linked to the field ID to the UntypedFormGroup', () => {
```
Comment (codaimaster) [inline projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/number/write-number-field.component.spec.ts:71]: UntypedFormGroup -> formGroup

File: projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/organisation/write-organisation-field.component.stories.ts [hunk 13]
```diff
@@ -93,13 +93,13 @@ export const standard: Story = template.bind({});
...
   parent: {
-    formGroup: {
-      organisationName: {
-        OrganisationID: 'Org1234',
-        OrganisationName: 'Waffles Ltd'
-      }
-    },
-    controls: {},
-    setControl: () => {}
+    formGroup {
```
Comment (codaimaster) [inline projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/organisation/write-organisation-field.component.stories.ts:96]: fix please

File: projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/palette.module.ts [hunk 14]
```diff
@@ -2,7 +2,7 @@ import { NgxMatDatetimePickerModule, NgxMatNativeDateModule, NgxMatTimepickerMod
 import { OverlayModule } from '@angular/cdk/overlay';
 import { CdkTreeModule } from '@angular/cdk/tree';
 import { CommonModule } from '@angular/common';
-import { ChangeDetectorRef, NgModule, Provider } from '@angular/core';
+import { ChangeDetectorRef, NgModule, Provider, NO_ERRORS_SCHEMA } from '@angular/core';
```
Comment (codaimaster) [inline projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/palette.module.ts:5]: Check implication.

File: projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/phone-uk/write-phone-uk-field.component.spec.ts [hunk 15]
```diff
@@ -65,7 +67,7 @@ describe('WritePhoneUKFieldComponent', () => {
     fixture.detectChanges();
   }));
 
-  it('should add a formControl linked to the field ID to the formGroup', () => {
+  it('should add a formControl linked to the field ID to the UntypedFormGroup', () => {
```
Comment (codaimaster) [inline projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/phone-uk/write-phone-uk-field.component.spec.ts:70]: UntypedFormGroup -> formGroup

File: projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/text-area/write-text-area-field.component.spec.ts [hunk 16]
```diff
@@ -64,7 +66,7 @@ describe('WriteTextAreaFieldComponent', () => {
     fixture.detectChanges();
   }));
 
-  it('should add a formControl linked to the field ID to the formGroup', () => {
+  it('should add a formControl linked to the field ID to the UntypedFormGroup', () => {
```
Comment (codaimaster) [inline projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/text-area/write-text-area-field.component.spec.ts:69]: UntypedFormGroup -> formGroup

File: projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/text/write-text-field.component.spec.ts [hunk 17]
```diff
@@ -65,7 +67,7 @@ describe('WriteTextFieldComponent', () => {
     fixture.detectChanges();
   }));
 
-  it('should add a formControl linked to the field ID to the formGroup', () => {
+  it('should add a formControl linked to the field ID to the UntypedFormGroup', () => {
```
Comment (codaimaster) [inline projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/text/write-text-field.component.spec.ts:70]: UntypedFormGroup -> formGroup

File: projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/yes-no/write-yes-no-field.component.spec.ts [hunk 18]
```diff
@@ -76,7 +78,7 @@ describe('WriteYesNoFieldComponent', () => {
     fixture.detectChanges();
   }));
 
-  it('should add a formControl linked to the field ID to the formGroup', () => {
+  it('should add a formControl linked to the field ID to the UntypedFormGroup', () => {
```
Comment (codaimaster) [inline projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/yes-no/write-yes-no-field.component.spec.ts:81]: UntypedFormGroup -> formGroup

File: projects/ccd-case-ui-toolkit/src/lib/shared/components/search-filters/search-filters.component.spec.ts [hunk 19]
```diff
@@ -539,7 +539,7 @@ describe('SearchFiltersComponent', () => {
         const arg: any = searchHandler.applyFilters.calls.mostRecent().args[0].selected;
         expect(arg['jurisdiction']).toEqual(JURISDICTION_3);
         expect(arg['caseType']).toEqual(CASE_TYPES_2[3]);
-        expect(arg['formGroup'].value).toEqual(formGroup.value);
+        expect(arg['UntypedFormGroup'].value).toEqual(formGroup.value);
```
Comment (codaimaster) [inline projects/ccd-case-ui-toolkit/src/lib/shared/components/search-filters/search-filters.component.spec.ts:542]: UntypedFormGroup -> formGroup

File: projects/ccd-case-ui-toolkit/src/lib/shared/components/workbasket-filters/workbasket-filters.component.spec.ts [hunk 20]
```diff
@@ -639,7 +639,7 @@ describe('with defaults', () => {
       });
       expect(arg['jurisdiction']).toEqual(JURISDICTION_2);
       expect(arg['caseType']).toEqual(CASE_TYPES_2[2]);
-      expect(arg['formGroup'].value).toEqual(formGroup.value);
+      expect(arg['UntypedFormGroup'].value).toEqual(formGroup.value);
```
Comment (codaimaster) [inline projects/ccd-case-ui-toolkit/src/lib/shared/components/workbasket-filters/workbasket-filters.component.spec.ts:642]: UntypedFormGroup -> formGroup

File: projects/ccd-case-ui-toolkit/src/lib/shared/directives/substitutor/label-substitutor.directive.spec.ts [hunk 21]
```diff
@@ -11,7 +11,7 @@ import createSpyObj = jasmine.createSpyObj;
 
 @Component({
   template: `
-    <tr ccdLabelSubstitutor [caseField]="caseField" [formGroup]="formGroup" [contextFields]="caseFields"
+    <tr ccdLabelSubstitutor [caseField]="caseField" [UntypedFormGroup]="UntypedFormGroup" [contextFields]="caseFields"
```
Comment (codaimaster) [inline projects/ccd-case-ui-toolkit/src/lib/shared/directives/substitutor/label-substitutor.directive.spec.ts:14]: UntypedFormGroup -> formGroup

File: projects/ccd-case-ui-toolkit/src/lib/shared/services/fields/fields.purger.ts [hunk 22]
```diff
@@ -112,9 +112,9 @@ export class FieldsPurger {
         for (const complexSubField of field.field_type.complex_fields) {
           if ((complexSubField.field_type.type === 'Complex' && complexSubField.field_type.complex_fields.length > 0) ||
             !complexSubField.retain_hidden_value) {
-            // Check for the existence of the parent FormGroup corresponding to the Complex field; if it exists, call
+            // Check for the existence of the parent UntypedFormGroup corresponding to the Complex field; if it exists, call
```
Comment (codaimaster) [inline projects/ccd-case-ui-toolkit/src/lib/shared/services/fields/fields.purger.ts:115]: UntypedFormGroup -> formGroup

File: projects/ccd-case-ui-toolkit/src/lib/shared/services/fields/fields.purger.ts [hunk 23]
```diff
@@ -147,11 +147,11 @@ export class FieldsPurger {
               }
 
               // Recursively delete the sub-field value if retain_hidden_value is false, OR if the sub-field type is
-              // Complex and regardless of retain_hidden_value, passing in the parent FormGroup
+              // Complex and regardless of retain_hidden_value, passing in the parent UntypedFormGroup
```
Comment (codaimaster) [inline projects/ccd-case-ui-toolkit/src/lib/shared/services/fields/fields.purger.ts:150]: UntypedFormGroup -> formGroup

File: projects/ccd-case-ui-toolkit/src/lib/shared/services/fields/fields.purger.ts [hunk 24]
```diff
@@ -190,12 +190,12 @@ export class FieldsPurger {
    * types, this recursive method is called until simple or "base" field types are reached. For `Document` field
    * types, its _sub-field_ `FormControl` values are set to null.
    *
-   * @param formGroup The `FormGroup` instance containing the `FormControl` for the specified field
+   * @param UntypedFormGroup The `UntypedFormGroup` instance containing the `FormControl` for the specified field
```
Comment (codaimaster) [inline projects/ccd-case-ui-toolkit/src/lib/shared/services/fields/fields.purger.ts:193]: UntypedFormGroup -> formGroup

File: projects/ccd-case-ui-toolkit/src/lib/shared/services/fields/fields.purger.ts [hunk 25]
```diff
@@ -220,9 +220,9 @@ export class FieldsPurger {
                   (field.retain_hidden_value
                     ? !complexSubField.retain_hidden_value || (parentField && !parentField.retain_hidden_value)
                     : true))) {
-                // The fieldControl is cast to a FormGroup because a Complex field type uses this as its underlying
+                // The fieldControl is cast to a UntypedFormGroup because a Complex field type uses this as its underlying
```
Comment (codaimaster) [inline projects/ccd-case-ui-toolkit/src/lib/shared/services/fields/fields.purger.ts:223]: a -> an

File: projects/ccd-case-ui-toolkit/src/lib/shared/services/fields/fields.purger.ts [hunk 26]
```diff
@@ -283,8 +283,8 @@ export class FieldsPurger {
           }
           break;
         case 'Document':
-          // NOTE: The field control (a FormGroup in this case) cannot just be set to null because Angular checks that
-          // all existing values of a FormGroup are present; setting the control's value to null causes a runtime error
+          // NOTE: The field control (a UntypedFormGroup in this case) cannot just be set to null because Angular checks that
+          // all existing values of a UntypedFormGroup are present; setting the control's value to null causes a runtime error
```
Comment (codaimaster) [inline projects/ccd-case-ui-toolkit/src/lib/shared/services/fields/fields.purger.ts:287]: a -> an

File: tsconfig.json [hunk 27]
```diff
@@ -12,10 +12,10 @@
     "experimentalDecorators": true,
     "target": "es2015",
```
Comment (codaimaster) [inline tsconfig.json:13]: check if es2020 could work better

**Review by OgunyemiO — APPROVED** [review]

**Review by RiteshHMCTS — APPROVED** [review]

**Review by andywilkinshmcts — DISMISSED** [review]

**Review by OgunyemiO — DISMISSED** [review]

**Review by RiteshHMCTS — DISMISSED** [review]

**Review by andywilkinshmcts — DISMISSED** [review]

**Review by OgunyemiO — APPROVED** [review]

**Pull request merged**